compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -873,6 +873,7 @@ impl<'a, 'b, 'tcx> BuildReducedGraphVisitor<'a, 'b, 'tcx> {
                     let msg = "macro-expanded `extern crate` items cannot \
                                        shadow names passed with `--extern`";
                     self.r.tcx.sess.span_err(item.span, msg);
+                    return;
```

Comment:
I think it's worth discussing the actual binding of `use std::xxx`.

When we add `return` here, `std::xxx` ultimately refers to `prelude::std`.

However, based on your [comment](https://github.com/rust-lang/rust/pull/111761#issuecomment-1557742102), it seems that `std::xxx` will point to `extern crate blash`(I guess, but not entirely sure).

There might be a difference between these two resolutions, so in my opinion, we should determine which action is better first.

Reply:
The choice is not that important because it only happens during failing compilation anyway.
The difference may only happen in some secondary diagnostics.

Reply:
If this case is migrated to regular ambiguity errors, then `extern crate blah` will be selected as a resolution, yes, because it's closer in scopes at the resolution point.
(This migration is probably a better long term solution, but it's a minor language change, so it's better done separately from fixing an ICE.)

Reply:
If the migration to regular ambiguity errors happens then this test case:
```rust
macro_rules! m {
    () => {
        extern crate core as std;
        
        use std::mem;
    }
}

m!();

fn main() {}
```
won't produce any errors.